src/server/scripts/Northrend/Naxxramas/instance_naxxramas.cpp

```diff
@@ -218,6 +224,23 @@ class instance_naxxramas : public InstanceMapScript
                         else
                             events.ScheduleEvent(EVENT_THADDIUS_RESET, 30 * IN_MILLISECONDS);
                         break;
+                    case EVENT_GLUTH_ZOMBIE_BEHAVIOR: // handles the change in the zombies' behavior after a decimate.
```

Comment:
Get a handle to them using SummonList, then handle behavior change in their AI. Don't do this in the instance script.